dashboard/pages/screening/callbacks.py

```diff
@@ -143,6 +145,11 @@ def on_outlier_purge_stage_entry(
     compounds_count = plates_count * bmg_vals.shape[2] * (bmg_vals.shape[3] - 2)
     outliers_count = (bmg_vals[:, 1] == 1).sum()
 
+    report_data = {}
+    report_data["plates_count"] = plates_count
```

Comment:
here you can already initialize the `report_data` dict with the counts, i.e.

```python
report_data = {
    "plates_count": plates_count,
    "compounds_count": compounds_count,
    "outliers_count": outliers_count
}
```

imo a bit cleaner